lib/backend/grpcclient/util.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2022 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package grpcclient
+
+import (
+	"github.com/gravitational/teleport/lib/backend"
+
+	bpb "github.com/gravitational/teleport/api/backend/proto"
+)
+
+func grpcItemsToBackendItems(in []*bpb.Item) []backend.Item {
```

Comment:
should this return `nil` if `in` is empty/nil?

Reply:
I don't _think_ so. As before, it seems to me that the implicit expectation is that `nil` be returned only on errors, and for empty responses an empty `backend.Whatever` is returned. I don't mind though, and I may be misinterpreting the expectations of backend implementations